### PR TITLE
feat(mentorship): add mentor My Programs page

### PR DESCRIPTION
**Pull request by Sameh16**
## Summary

- Adds the mentor **My Programs** landing page at `/mentorship/mentor/programs`, with underline tabs for **My Programs** and a placeholder **Mentor Profile** tab.
- Introduces a mentor program card that mirrors the admin program card layout (avatar, org/term line, term status badge, title, metrics, chevron) with mentor-specific stats: Mentees, Tasks to Review, and Applicants.
- Wires a mock-backed BFF endpoint (`GET /api/mentorship/mentor/programs`) and frontend `MentorshipService.getMentorPrograms()` to load the list.
- Adds a temporary **My Programs** button on the Become a Mentor registration page for navigation until the profiles API can route mentors automatically.
- Program cards are clickable and currently show a coming-soon toast; program detail navigation will follow in a later change.

## Test plan

- [ ] Navigate to `/mentorship/mentor` and click **My Programs** — lands on `/mentorship/mentor/programs`
- [ ] Confirm the page title, tab count badge (after load), and program cards render with correct metrics
- [ ] Click a program card — coming-soon toast appears
- [ ] Switch to **Mentor Profile** tab — placeholder empty state renders
- [ ] Use keyboard (Arrow keys) to move between tabs
- [ ] Run unit tests: `yarn test` (covers `MentorProgramsComponent` and `MentorProgramCardComponent`)

### CODE REVIEW

**Comment by dealako** [comment]
Hi @Sameh16 :wave: thanks for this. Clean, well-scoped feature: the mentor **My Programs** page mirrors the admin card layout nicely, the tab a11y wiring (roles, `aria-selected`, roving `tabindex`, arrow-key handling) is thoughtful, and threading everything through `@lfx-one/shared` types/constants plus the `MentorshipService` error-degrade pattern keeps it consistent with the existing mentorship surfaces. Mock-backed and read-only, so blast radius is small. The items below are all polish, correctness of test coverage, and one forward-looking backend note, nothing structural.

I ran a multi-dimensional pass (security, correctness/perf/tests, style/api/docs) and reconciled with the existing @Copilot inline comments. Agreeing with all four of Copilot's points below.

### 🟡 Minor: 4
1. **Loading gate hides the Profile panel** (`mentor-programs.component.html:44`): the `!hasLoaded()` branch wraps *all* tab content, so selecting **Mentor Profile** while the programs request is pending shows the programs loader instead of the profile panel, and the Profile tab's `aria-controls` points to a panel not in the DOM. Branch on `activeTab()` first and keep the loader inside the programs tabpanel. Agree with [Copilot](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008692707).
2. **Tailwind class map not safelisted** (`mentorship-mentor.constants.ts:26`): `MENTORSHIP_MENTOR_PROGRAM_TERM_STATUS_BADGE_CLASSES` lives in `@lfx-one/shared` (outside the `content` glob) but isn't spread into `apps/lfx-one/tailwind.config.js` alongside the sibling `MENTORSHIP_*` maps. It renders today only because every token (`bg-blue-50`, `text-amber-700`, etc.) happens to appear as a literal elsewhere; a future palette change silently loses styling. Add it to the safelist as its own source of truth. Agree with [Copilot](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008692881).
3. **Card keyboard handlers untested** (`mentor-program-card.component.spec.ts:55`): the template wires `(keydown.enter)` and `(keydown.space)` on the card, but the spec only exercises `.click()`. The keyboard paths (and the Space `preventDefault`) are unverified. See inline comment.
4. **Outage indistinguishable from empty** (`mentorship.service.ts:53`): `getMentorPrograms` degrades every failure to an empty response, so the page renders `0` + "No programs yet" on an auth/network fault. This is the intentional repo-wide degrade pattern (mirrors `getPrograms`/`getLfProjects`), so it's not wrong, but a mentor's *own* landing page arguably warrants an explicit error/retry state. Agree with [Copilot](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008692831); flagging as a product call.

### ⚪ Nit: 1
1. **ArrowRight test asserts panel, not focus** (`mentor-programs.component.spec.ts`): "moves focus across tabs with ArrowRight" only checks the profile panel rendered, not that the tab button received `document.activeElement`. Removing the `.focus()` call would still pass. Agree with [Copilot](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008692785).

### ❔ Question: 1
1. **Per-user scoping for "my" programs** (`server/services/mentorship.service.ts:89`): `getMentorPrograms` returns the full mock list without reading the caller's identity. Fine for a mock, but this is a "my programs" surface. When the real backend lands, is the plan to filter by the authenticated mentor? Worth threading the username through now so the ownership-check seam exists before it's DB-backed. See inline comment.

### Decision
🔴 **Needs changes before approval** — no blocking or privacy issues, and nothing security-critical (mock-backed, read-only, behind existing auth). Requesting changes on the basis of the minor set (loading-gate a11y bug + safelist + test-coverage gaps). Most of these are quick; the error-state and scoping items are product/roadmap calls you can resolve by comment.

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/components/mentor-program-card/mentor-program-card.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    expect(metrics?.textContent).toContain('Mentees');
+    expect(metrics?.textContent).toContain('3');
+    expect(metrics?.textContent).toContain('Tasks to Review');
+    expect(metrics?.textContent).toContain('4');
+    expect(metrics?.textContent).toContain('Applicants');
+    expect(metrics?.textContent).toContain('5');
+  });
+
+  it('emits the program id when the card is clicked', () => {
```
Comment (dealako) [inline apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/components/mentor-program-card/mentor-program-card.component.spec.ts:55]: [minor] Card keyboard handlers are untested

Issue: The card template wires `(keydown.enter)="onCardClick()"` and `(keydown.space)="onCardClick(); $event.preventDefault()"`, but this spec only exercises `.click()`. The Enter/Space activation paths and the Space `preventDefault` (which stops the page from scrolling) are never verified.

Proof: This is the only interaction test in the card spec and it calls `?.click()` on the card element. No `KeyboardEvent` is dispatched anywhere in the file.

Why it matters: The card is a `role="button"` `tabindex="0"` div, so keyboard activation is the accessibility contract, not a nicety. A regression that drops either handler would pass CI.

Fix: Add two cases, e.g.
```ts
it('emits on Enter', () => {
  const onClick = vi.fn();
  fixture.componentInstance.cardClick.subscribe(onClick);
  const card = element().querySelector<HTMLElement>('[role="button"]')!;
  card.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter', bubbles: true }));
  expect(onClick).toHaveBeenCalledWith(mentorProgram.id);
});
```
and a Space case that also asserts `preventDefault` fired.

File: apps/lfx-one/src/server/services/mentorship.service.ts [hunk 2]
```diff
@@ -84,6 +86,13 @@ export class MentorshipService {
     return page;
   }
 
+  public async getMentorPrograms(req: Request): Promise<MentorshipMentorProgramsResponse> {
```
Comment (dealako) [inline apps/lfx-one/src/server/services/mentorship.service.ts:89]: [question] Per-user scoping for a "my programs" surface

Issue: `getMentorPrograms(req)` returns the full `MOCK_MENTORSHIP_MENTOR_PROGRAMS` list and never reads the caller's identity, even though the controller already resolves `getUsernameFromAuth(req)` at the route boundary.

Proof: `const data = MOCK_MENTORSHIP_MENTOR_PROGRAMS.map((program) => ({ ...program }));` — no filter by mentor identity.

Why it matters: Correct for a mock, but this is the "My Programs" endpoint. If the real backend swap keeps this signature and simply replaces the array with a store query, an unfiltered query would expose every mentor's programs and applicant/mentee counts (broken object-level authorization by omission).

Fix (non-blocking): thread the authenticated username into the service now, e.g. `getMentorPrograms(req, username)`, and filter the mock by it so the ownership-check seam exists before it's DB-backed. If scoping is deliberately deferred, a `// TODO: scope to authenticated mentor` note here would capture the intent.

**Review by dealako — CHANGES_REQUESTED** [review]
Requesting changes on the minor set (loading-gate a11y bug, Tailwind safelist omission, test-coverage gaps). No blocking, privacy, or security-critical findings. Full breakdown and reconciliation with @Copilot's comments is in the summary comment. Two net-new inline notes below; the rest agree with Copilot's existing threads.

**Comment by Sameh16** [comment]
Thanks a ton for the thorough review @dealako (and the extra passes from Copilot & Bugbot)! Here's a rundown of what was addressed:

- ✅ **Loading-gate a11y bug** — restructured the template to branch on `activeTab()` first, so the Mentor Profile tab stays mounted and reachable while programs are still loading.
- ✅ **Keyboard-navigation test coverage** — the tab test now asserts `document.activeElement` and `event.defaultPrevented`, so removing the `.focus()` call would fail CI.
- ✅ **Silent error fallback** — the frontend service now rethrows and the component surfaces a proper error state with a Retry action instead of collapsing to “No programs yet”.
- ✅ **Tailwind safelist omission** — `MENTORSHIP_MENTOR_PROGRAM_TERM_STATUS_BADGE_CLASSES` is now imported and spread into `apps/lfx-one/tailwind.config.js` alongside the neighboring mentorship maps.
- ✅ **Card keyboard-activation tests** — added Enter and Space cases; the Space test also asserts `preventDefault` fired.
- 💬 **Per-user scoping on `getMentorPrograms`** — since this is mock data only and the entire method will be replaced when the mentorship-service is wired up, we're intentionally not adding a filter seam here. The controller keeps the `getUsernameFromAuth(req)` auth boundary in place for when the real backend lands.

Ready for another look whenever you have a minute 🙏

File: packages/shared/src/constants/mentorship-mentor.constants.ts [hunk 3]
```diff
@@ -1,8 +1,145 @@
...
+  'active-term': 'Active term',
+  upcoming: 'Upcoming',
+  completed: 'Completed',
+};
+
+export const MENTORSHIP_MENTOR_PROGRAM_TERM_STATUS_BADGE_CLASSES: Record<MentorshipMentorProgramTermStatus, string> = {
+  'active-term': 'bg-blue-50 text-blue-700',
+  upcoming: 'bg-amber-50 text-amber-700',
+  completed: 'bg-gray-100 text-gray-600',
+};
```
[the remark this answers is not in the document]
Comment (Sameh16) [reply packages/shared/src/constants/mentorship-mentor.constants.ts:30]: Good catch — fixed. `MENTORSHIP_MENTOR_PROGRAM_TERM_STATUS_BADGE_CLASSES` is now imported and spread into `apps/lfx-one/tailwind.config.js` alongside the neighboring mentorship class maps, so it owns its safelist entry independently of what the sibling maps happen to overlap with. Thanks!

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/components/mentor-program-card/mentor-program-card.component.spec.ts (hunk 1 above)
Comment (Sameh16) [reply apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/components/mentor-program-card/mentor-program-card.component.spec.ts:55]: Great point — addressed. Added two new cases in `mentor-program-card.component.spec.ts`:

- **Enter** — dispatches a `KeyboardEvent('keydown', { key: 'Enter' })` on the card and asserts `cardClick` emits with the program id.
- **Space** — dispatches `{ key: ' ', code: 'Space' }` and asserts both the emission *and* `event.defaultPrevented === true`, so a regression that drops the `$event.preventDefault()` for Space would fail CI.

The click case still runs alongside them, so all three activation paths are locked in. Thanks!

File: apps/lfx-one/src/server/services/mentorship.service.ts (hunk 2 above)
Comment (Sameh16) [reply apps/lfx-one/src/server/services/mentorship.service.ts:89]: Good call-out. Intentionally deferring this one — since the current implementation is mock data only and the entire method body will be replaced when the mentorship-service is wired up, threading `username` through now would just be scaffolding around a filter we won't use.

The controller already resolves `getUsernameFromAuth(req)` at the route boundary, so the auth seam is in place; the real backend swap will introduce the ownership filter as part of that change (not a `MOCK_*` slice). Happy to add a `// TODO` note if you'd prefer the intent captured in-file.

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        }),
+        switchMap(() =>
+          this.mentorshipService.getMentorPrograms().pipe(
+            map((response) => {
+              this.hasLoaded.set(true);
+              return response;
+            }),
+            catchError((error: HttpErrorResponse) => {
+              this.hasLoaded.set(true);
+              this.loadError.set(typeof error.error?.message === 'string' ? error.error.message : 'We could not load your mentor programs. Please retry.');
```
[the remark this answers is not in the document]
Comment (Sameh16) [reply apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts]: Acknowledging as a known limitation of the shared error-transport shape (matches the learned-rule suppression Bugbot flagged inline). The subtitle intentionally falls through to the friendly copy when the server-authored `message` isn't present; wiring up the transport-string fallback is a cross-cutting change that belongs in the shared error helper, not this caller. Not blocking for this PR.

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  private readonly platformId = inject(PLATFORM_ID);
+  private readonly tabBtns = viewChildren<ElementRef<HTMLButtonElement>>('tabBtn');
+
+  protected readonly title = MENTORSHIP_MENTOR_PROGRAMS_PAGE_TITLE;
+  protected readonly tabs = MENTORSHIP_MENTOR_PAGE_TABS;
+  protected readonly activeTab = signal<MentorshipMentorPageTab>('programs');
+  protected readonly hasLoaded = signal(false);
+  protected readonly loadError = signal<string | null>(null);
+
+  private readonly reloadPrograms = signal(0);
```
[the remark this answers is not in the document]
Comment (Sameh16) [reply apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts:42]: Good catch — will gate the tab count on `!loadError()` as well so the badge stays hidden when the load fails:

```ts
count: tab.value === 'programs' && this.hasLoaded() && !this.loadError() ? this.programCount() : null,
```

Rolling this into the next commit on the branch.

**Comment by dealako** [comment]
Thanks for the fast turnaround @Sameh16 :wave: this revision closes out every actionable item from the last round.

### 👏 Nice work
- **Loading gate fixed** — the template now branches on `activeTab()` first and keeps the loader *inside* the programs tabpanel, so the Mentor Profile panel is always mounted and its `aria-controls` target exists during load. Exactly the restructure I was hoping for.
- **Explicit error + retry state** — the new `loadError` signal, `reloadPrograms` trigger, and the dedicated error empty-state (with a Retry CTA) cleanly separate "outage" from "genuinely no programs." Switching `getMentorPrograms` to `rethrowError` so the component owns the UX is the right call.
- **Test coverage landed** — new specs cover the error-propagation path (`service.spec` + `component.spec`), both card keyboard handlers (Enter, and Space asserting `defaultPrevented === true`), and the ArrowRight test now asserts `document.activeElement` + `defaultPrevented`, not just panel render.
- **Tailwind safelist** — `MENTORSHIP_MENTOR_PROGRAM_TERM_STATUS_BADGE_CLASSES` is now imported and spread into the safelist following the established pattern.

### Revision status
- ✅ Loading gate hides Profile panel — **resolved** ([commit 7977438](https://github.com/linuxfoundation/lfx-self-serve/commit/7977438c5044225ea2eeda445594fc10eb0b38aa))
- ✅ Card keyboard handlers untested — **resolved**
- ✅ Outage indistinguishable from empty — **resolved**
- ✅ Tailwind class map not safelisted — **resolved**
- ✅ ArrowRight test asserts panel not focus — **resolved**
- ❔ Per-user scoping (`server/services/mentorship.service.ts` `getMentorPrograms`) — **still open**, but this is the forward-looking mock-backend question from last round, not a blocker.

### 🟡 Minor: 2 (both new, both flagged by @cursor)
1. **Dead BFF error-message path** (`mentor-programs.component.ts:99`): `error.error?.message` only reads one of the two BFF error shapes. The error-class proxy path (`BaseApiError#toResponse`) puts the text under `error`, so the server-authored reason is silently dropped and the retry state always shows generic copy. The repo already solved this — `serverAuthoredMessage` / `extractErrorMessage` in `shared/utils/http-error.utils.ts` (GH-2349) handle both shapes. See inline. Agree with [@cursor](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008849147).
2. **Zero-count badge during error** (`mentor-programs.component.ts:96-99`): the error branch sets `hasLoaded(true)` and returns the empty response, so the My Programs tab badge renders `0` while the panel says "Could not load your programs." Suppress the count when `loadError()` is set (e.g. `count: tab.value === 'programs' && hasLoaded() && !loadError() ? programCount() : null`). Agree with [@cursor](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008849158).

### Decision
🔴 **Needs changes before approval** — no blocking, privacy, or security issues; this is purely the two minor error-state polish items above. Both are small and localized to the error path you just added. The dead-message path is worth fixing since it defeats the purpose of the server-authored reason this commit set out to surface. Once those land (and optionally a `// TODO` on the server for per-user scoping), this is good to go.

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts (hunk 4 above)
Comment (dealako) [inline apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts]: [minor] Dead BFF error-message path

Issue: `error.error?.message` reads only one of the two error shapes this BFF emits. The error-class proxy path (`BaseApiError#toResponse`) carries the server-authored text under `error` (a string, or `error.error`), not `error.message`. On that path the ternary falls through to the generic fallback, so the reason the server wrote never reaches the retry state.

Proof: `shared/utils/http-error.utils.ts` documents exactly this (GH-2349): `serverAuthoredMessage`/`extractErrorMessage` exist because "the error-class proxy path carries the text under `error`, where `err.error?.message` reads nothing." `hasServerAuthoredMessage` handles string bodies, `{error}`, `{message}`, and `{errors[].message}`. This new code reimplements a strictly narrower subset and loses the error-class shape.

Why it matters: this commit's stated goal is to surface the server's reason in the retry state; on the error-class path it silently shows generic copy instead, so the feature is half-wired.

Fix: use the existing helper instead of a hand-rolled read:
```ts
import { serverAuthoredMessage } from '@shared/utils/http-error.utils';
// ...
catchError((error: HttpErrorResponse) => {
  this.hasLoaded.set(true);
  this.loadError.set(serverAuthoredMessage(error, 'We could not load your mentor programs. Please retry.'));
  return of(EMPTY_MENTORSHIP_MENTOR_PROGRAMS_RESPONSE);
})
```
(Note: the @cursor learned-rule exemption about the `extractErrorMessage` transport-string fallback doesn't apply here — this code doesn't call that helper at all.)

**Review by dealako — CHANGES_REQUESTED** [review]
Follow-up review of commit 7977438. Every actionable item from the prior round is resolved (loading gate, error/retry state, keyboard tests, tailwind safelist, focus assertion). Requesting changes only on the two minor error-state polish items introduced by this commit, both also flagged by @cursor. Full breakdown in the summary comment. One net-new inline note below adds the repo-helper reference @cursor didn't cite.

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts (hunk 4 above)
Comment (Sameh16) [reply apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts]: You're absolutely right — my earlier note misread the exemption. The hand-rolled `error.error?.message` never called into `extractErrorMessage`, so the learned rule didn't apply.

Switched to the shared helper:

```ts
import { serverAuthoredMessage } from '@app/shared/utils/http-error.utils';
// ...
catchError((error: HttpErrorResponse) => {
  this.hasLoaded.set(true);
  this.loadError.set(serverAuthoredMessage(error, 'We could not load your mentor programs. Please retry.'));
  return of(EMPTY_MENTORSHIP_MENTOR_PROGRAMS_RESPONSE);
})
```

Now both the `HttpErrorResponse.error.message` path *and* the `BaseApiError#toResponse` string/`{error}` shape surface the server-authored reason, with the friendly copy only kicking in when nothing usable is present. Thanks for the pushback — much cleaner.

**Comment by Sameh16** [comment]
Thanks for the fast turnaround on the second pass @dealako 🙏 Both new items are addressed:

- ✅ **Dead BFF error-message path** — swapped the hand-rolled `error.error?.message` read for `serverAuthoredMessage(error, fallback)` from `@app/shared/utils/http-error.utils`, so both the `HttpErrorResponse` shape and the `BaseApiError#toResponse` shape surface the server-authored reason in the retry state. You were right that the learned-rule exemption didn't apply here — my earlier reply misread it.
- ✅ **Zero-count badge during error** — the `tabItems` computed now gates on `!loadError()`, so the My Programs badge stays hidden while the error empty-state is showing:
  ```ts
  count: tab.value === 'programs' && this.hasLoaded() && !this.loadError() ? this.programCount() : null,
  ```

Per-user scoping is still deferred (mock-only, entire method body gets replaced when the mentorship-service lands), but happy to add a `// TODO` in-file if you'd like the intent captured.

Ready for another look whenever you have a minute!

**Comment by Sameh16** [comment]
Quick note on the 5 unit-test failures in the last CI run — they are **not related to this PR**.

All 5 failures are in `OrgEasyclaDetailComponent — the approval tab` describe block:

```
spec-app-modules-dashboards-org-org-easycla-org-easycla-detail-org-easycla-detail.component.spec.js
  > renders the approval list panel when the tab is selected
  > does not load the approval list until the tab is opened
  > shows the row count until the tab reports its own
  > takes the count the tab reports after a write
  > shows a dash when the row carries no count

TypeError: Cannot read properties of undefined (reading 'pipe')
  at _OrgEasyclaDetailComponent.<instance_members_initializer>
    org-easycla-detail.component.ts:143:30
```

### Why it isn't this PR

- The failing component is in the **org-easycla** module. This PR only touches the **mentorship** module. There is no shared code path between them (no shared service, no overlapping DI, no shared route).
- The failing tests were freshly rewritten on `main` in [#2394 (`83a85e9`)](https://github.com/linuxfoundation/lfx-self-serve/commit/83a85e9e7e476e66386b5aeff234aa3092520a64) two minutes before this PR opened. That PR re-keyed the detail route from `:signatureId` to `:claGroupId` and reworked the initializer block where the crash originates (`previewSelection`, `previewing`, `selectedOrgUid$`).
- The failure is a class-field-initialization crash — the component fails to instantiate at all, which is upstream of any test assertion this PR could influence.

### Next step

I've merged latest `main` into this branch (HEAD is now `30920ba`). No further fix for the easycla failure has landed on `main` since the failing run, so this looks like a flaky Vitest isolation edge case. Re-running the failed job on the new HEAD should clear it. If the same 5 tests fail deterministically again, I'll ping `@ahmedomosanya` since it would then be a real issue with #2394's initializer race, not this PR's changes.

**Comment by dealako** [comment]
Nice, this is dialed in now @Sameh16 :tada: both items from the last round are fixed exactly as suggested, and the fixes are correct on trace-through.

### 👏 Nice work
- **Dead BFF error-message path resolved** — swapping `error.error?.message` for `serverAuthoredMessage(error, fallback)` ([`shared/utils/http-error.utils.ts`](https://github.com/linuxfoundation/lfx-self-serve/blob/81603541bc0231d803244c5351a233d801121ae1/apps/lfx-one/src/app/shared/utils/http-error.utils.ts)) now surfaces both BFF error shapes (`{error}` error-class proxy path and `{message}` direct path), so the server-authored reason actually reaches the retry state. Exactly the helper for the job.
- **Zero-count badge resolved** — the `&& !this.loadError()` guard on the `tabItems` count cleanly suppresses the `0` badge during the error state; on success `loadError` stays null and the count shows. Signal reactivity tracks correctly.

### Revision status
- ✅ Dead BFF error-message path — **resolved** ([commit 8160354](https://github.com/linuxfoundation/lfx-self-serve/commit/81603541bc0231d803244c5351a233d801121ae1))
- ✅ Zero-count badge during error — **resolved**
- ✅ (round 1) loading gate, keyboard tests, outage state, tailwind safelist, focus assertion — all still resolved, no regressions
- ❔ Per-user scoping (`server/services/mentorship.service.ts`) — **still open**, unchanged; the forward-looking mock-backend question, not a blocker for this PR.

### 🟡 Minor: 2
1. **Error-state badge suppression not asserted** (`mentor-programs.component.spec.ts:151`): the error test verifies the error panel + retry, but doesn't assert the `My Programs` count badge is *absent* while `loadError()` is set — which is the exact behavior this commit added. A future refactor could reintroduce the `0` badge without failing CI. Add an assertion that the tab's count span is gone in the error state.
2. **Empty-state copy points to an unavailable action** (`mentor-programs.component.html:75`, pre-existing, not in this commit): "Request to join a program from your mentor profile to get started." sends users to the Mentor Profile tab, which is still a placeholder in this PR, and the registration copy says joining isn't available yet. Neutral copy until that workflow ships avoids the dead end. Agree with [@Copilot](https://github.com/linuxfoundation/lfx-self-serve/pull/2408#discussion_r4008849147).

### ⚪ Nit: 1
1. **Import alias + placement** (`mentor-programs.component.ts:18`): uses `@app/shared/utils/http-error.utils` and sits below the relative imports. Every other importer of this module uses the more specific `@shared/utils/http-error.utils` alias, grouped with the other `@`-alias imports at the top. No lint rule enforces it, purely consistency. `import { serverAuthoredMessage } from '@shared/utils/http-error.utils';` moved up with the others.

### Decision
✅ **Approved with minor comments** — every behavioral finding from both prior rounds is resolved, and nothing here is blocking, privacy, or security-related. The two minors are a test assertion and a pre-existing copy tweak; the nit is import style. None need to gate the merge — feel free to fold them in on your next push or a quick follow-up. Thanks for the tight iteration on this. :rocket:

File: apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+import { EmptyStateComponent } from '@components/empty-state/empty-state.component';
+import { RouteLoadingComponent } from '@components/loading/route-loading.component';
+import { EMPTY_MENTORSHIP_MENTOR_PROGRAMS_RESPONSE, MENTORSHIP_MENTOR_PAGE_TABS, MENTORSHIP_MENTOR_PROGRAMS_PAGE_TITLE } from '@lfx-one/shared/constants';
+import { MentorshipMentorPageTab, MentorshipMentorProgramsResponse } from '@lfx-one/shared/interfaces';
+import { MentorshipService } from '@services/mentorship.service';
+import { catchError, map, of, switchMap, tap } from 'rxjs';
+
+import { MentorshipComingSoonService } from '../../services/mentorship-coming-soon.service';
+import { MentorProgramCardComponent } from './components/mentor-program-card/mentor-program-card.component';
+import { serverAuthoredMessage } from '@app/shared/utils/http-error.utils';
```
Comment (dealako) [inline apps/lfx-one/src/app/modules/mentorship/mentor/mentor-programs/mentor-programs.component.ts:18]: [nit] Import alias + placement

Issue: This imports from `@app/shared/utils/http-error.utils` and sits below the two relative imports, separated from the other `@`-alias imports at the top of the file.

Proof: Every other importer of this module in the app uses the more specific `@shared/utils/http-error.utils` alias (e.g. `account-settings.component.ts:21`, plus the service files), and groups it with the top `@`-alias block. Both aliases resolve (tsconfig maps `@app/*`->./src/app/* and `@shared/*`->./src/app/shared/*), and no eslint import-order rule enforces the grouping, so this is purely consistency.

Why it matters: matches the established convention and keeps the import block scannable.

Fix: `import { serverAuthoredMessage } from '@shared/utils/http-error.utils';` moved up with the other `@`-alias imports (after the `@services` line).

**Review by dealako — APPROVED** [review]
Approved with minor comments. Both findings from the prior round are resolved correctly (serverAuthoredMessage swap + !loadError() count guard), traced through and confirmed. Remaining items are non-blocking: a test-coverage assertion, a pre-existing empty-state copy tweak (agreeing with @Copilot), and one import-style nit below. Full breakdown in the summary comment. The forward-looking per-user-scoping question on the server service stays open but doesn't gate this PR. Nice iteration.
